### PR TITLE
Acquisition total comments

### PR DESCRIPTION
**Pull request by gforcada**
Fixes PR #58 that was mostly a proof of concept and room for discussion.

@tisto does this look better?

(sorry for PR #60, seems that github does not allow you to change the branch you base your pull request once you create it...)

### CODE REVIEW

**Comment by gforcada** [comment]
@tisto I made travis run on 2.2.x branch, just to realize that is only running code-analysis on it.

I see *lots* of errors on travis output, see https://travis-ci.org/plone/plone.app.discussion/builds/50489435 as an example.

**Comment by gforcada** [comment]
@tisto or anyone else, I think this is good to go, it can be both merged in 2.2.x and once done I will create another pull request for master.

File: plone/app/discussion/conversation.py [hunk 1]
```diff
@@ -89,8 +89,22 @@ def enabled(self):
...
+    def public_comments(self):
+        """Count total comments.
+
+        The @property decorator used on ``total_comments`` method doesn't work
+        when combined with acquisition, that's a fact of life.
+
+        Thus the total_comments indexer (on p.a.discussion.catalog.py) fails
+        to index properly if acquisition is involved.
+
+        To solve it, this method takes over the indexing and is being used by
```
Comment (tisto) [inline plone/app/discussion/conversation.py]: @gforcada Why don't we just replace the total_comments implementation? What kind of bbb scenario do you have in mind?

Comment (gforcada) [inline plone/app/discussion/conversation.py]: Well, any usage of p.a.discussion like:

```python
conversation = IConversation(self.context)
return conversation.total_comments
```

Will need to be replaced for:

```python
conversation = IConversation(self.context)
return conversation.public_comments()
```

Or if we do replace it, parenthesis need to be appended to ``total_comments``.

On our freitag.de packages for example that means to at least 4 to 5 lines having to be changed. That's completely fine for me, I don't know what Plone's policy is for backwards compatibility at method level on a stable (4.3.x version) release.

For 5.0 I would say that yes, let's change it. For 4.3.4 (as this PR is about) maybe is not such a good idea to break backwards compatibility on a micro release (what will end up being 4.3.5).

Comment (tisto) [inline plone/app/discussion/conversation.py]: Right. Forgot about the @property.

**Pull request merged**